packages/dotnet-sdk/src/TeamsFx/Conversation/ConversationBot.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace Microsoft.TeamsFx.Conversation
+{
+    using Microsoft.Bot.Builder;
+
+    /// <summary>
+    /// Provide utilities for bot conversation, including:
+    /// <list type="bullet">
+    ///     <item>
+    ///         <description>send notification to varies targets (e.g., member, group, channel).</description>
+    ///     </item>
+    /// </list>
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Set <c>Adapter</c> in <see cref="ConversationOptions"/> to use your own bot adapter.
+    /// </para>
+    /// <para>
+    /// For notification, set <c>Notification.Storage</c> in <see cref="ConversationOptions"/> to use your own storage implementation.
+    /// </para>
+    /// </remarks>
+    public class ConversationBot
+    {
+        /// <summary>
+        /// The bot adapter.
+        /// </summary>
+        public BotAdapter Adapter { get; private set; }
```

Comment:
So the adapter is inited outside the SDK,  I mean user created it and put it as an input?

Reply:
We'll generate the initializing part in template, so user can still focus on business logic.

Reply:
OK, got you. Now the CloudAdapter is using the BotFrameworkAuthentication, it forced the appSettings to a fixed key name -- MicrosoftAppId, a very confusing name.  
What in my mind is put the init logic into SDK, user just using BOT_ID and BOT_PASSOWRD as input to our SDK. Like what we delivered in js SDK.  
Just an idea of ​​mine, you can think about it in the future

Reply:
I've tried following code to create BotFrameworkAuthentication:
``` csharp
// Prepare Configuration for ConfigurationBotFrameworkAuthentication
builder.Configuration["MicrosoftAppType"] = "MultiTenant";
builder.Configuration["MicrosoftAppId"] = builder.Configuration.GetSection("BOT_ID")?.Value;
builder.Configuration["MicrosoftAppPassword"] = builder.Configuration.GetSection("BOT_PASSWORD")?.Value;

// Create the Bot Framework Authentication to be used with the Bot Adapter.
builder.Services.AddSingleton<BotFrameworkAuthentication, ConfigurationBotFrameworkAuthentication>();
```

Then we can leverage the built-in logic of `ConfigurationBotFrameworkAuthentication` such as client cache and credential. But I feel like these codes should be put in template, not SDK.

Reply:
The key point is, if we put the init logic in user's code, it's easy for user to integrate with ASP.Net Core's DI (Service Provider).
But if we put the init logic in SDK, we actually hide the DI part so it's a bit complex for user to integrate with service provider.

Reply:
Agreed! Let the authentication part outside SDK provide more ways for user to use.
Just a question, outside this pr. Does our js SDK has same logic or this is the different between these two SDK. (because I saw js SDK just put id and password as input)

Reply:
That's another story...
In js sdk we use `BotFrameworkAdapter` whose initialization is simple, so we are able to directly use id and password.
But in .net sdk, `BotFrameworkAdapter` is **obsoleted** so we need to move to `CloudAdapter`, whose initialization is a bit complex.

I have no idea why js `BotFrameworkAdapter` is not obsoleted. But if later we need to move to `CloudAdapter` in js as well, we may change the js initialization logic as well.